src/config/marinade-config.ts

```diff
@@ -6,15 +6,20 @@ export class MarinadeConfig {
   marinadeFinanceProgramId = new web3.PublicKey('MarBmsSgKXdrN1egZf5sqe1TMai9K1rChYNDJgjq7aD')
   marinadeReferralProgramId = new web3.PublicKey('mRefx8ypXNxE59NhoBqwqb3vTvjgf8MYECp4kgJWiDY')
 
+  // marinade instance public key
   marinadeStateAddress = new web3.PublicKey('8szGkuLTAux9XMgZ2vtY39jVSowEcpBfFfD8hXSEqdGC')
+
   marinadeReferralGlobalStateAddress = new web3.PublicKey('mRg6bDsAd5uwERAdNTynoUeRbqQsLa7yzuK2kkCUPGW')
 
   stakeWithdrawAuthPDA = new web3.PublicKey('9eG63CdHjsfhHmobHgLtESGC8GabbmRcaSpHAZrtmhco')
 
   connection = new web3.Connection(DEFAULT_PROVIDER_URL)
   publicKey: web3.PublicKey | null = null
 
+  // referral PDA state account
   referralCode: web3.PublicKey | null = null
+  // partner associated token account (ATA)
+  tokenPartnerAccount: web3.PublicKey | null = null
```

Comment:
can the partner account be read from the referralAccount, and then the ATA computed?
meaning we only need the referralCode

Reply:
ok, no, we store mSolPartnerAccount to easily validate in the instructions.
We only verify that mSolPartnerAccount.owner == partnerAccount and 
mSolPartnerAccount.mint == mSOL.mint (constant)